### PR TITLE
CORE-4163: Move extensions and helpers out of membership-impl lib

### PR DESCRIPTION
**Pull request by YashNabar**
https://r3-cev.atlassian.net/browse/CORE-4163
`MemberInfoExtension.kt` and `MemberInfoHelpers.kt` moved to `membership-common` to remove dependency on `membership-impl` in non-membership modules.

### CODE REVIEW

File: components/membership/membership-group-read-impl/test.bndrun [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 -runrequires: \
     bnd.identity;id='net.corda.membership-group-read-impl',\
+    bnd.identity;id='net.corda.membership-impl',\
```
Comment (charlieR3) [inline components/membership/membership-group-read-impl/test.bndrun:18]: Why is that required now but not before?

File: components/membership/membership-persistence-service-impl/build.gradle [hunk 2]
```diff
@@ -46,7 +45,6 @@ dependencies {
     integrationTestImplementation project(':libs:configuration:configuration-datamodel')
     integrationTestImplementation project(':libs:db:db-admin')
     integrationTestImplementation project(':libs:messaging:topic-admin')
-    integrationTestImplementation project(':libs:membership:membership-impl')
```
Comment (charlieR3) [inline components/membership/membership-persistence-service-impl/build.gradle:50]: Is it possible to remove it from `components/membership/membership-persistence-service-impl/test.bndrun` also if it's no longer a runtime dependency for running the integration tests?

File: libs/membership/ledger-identity/build.gradle [hunk 3]
```diff
@@ -10,6 +10,7 @@ dependencies {
     compileOnly 'org.osgi:org.osgi.service.component.annotations'
 
     implementation platform("net.corda:corda-api:$cordaApiVersion")
+    implementation project(':libs:membership:membership-common')
     implementation project(':libs:membership:membership-impl')
```
Comment (charlieR3) [inline libs/membership/ledger-identity/build.gradle]: Is `membership-impl` still required here?

File: components/membership/membership-persistence-service-impl/build.gradle (hunk 2 above)
Comment (charlieR3) [reply components/membership/membership-persistence-service-impl/build.gradle:50]: Spoke offline. Member info factory impl still needed for integration tests

File: libs/membership/ledger-identity/build.gradle (hunk 3 above)
Comment (YashNabar) [reply libs/membership/ledger-identity/build.gradle]: Split into impl dependency on `layered-property-map` and test dependency on `membership-impl`

**Review by charlieR3 — APPROVED** [review]
